Monika After Story/game/script-ch30.rpy

```diff
@@ -1679,21 +1679,39 @@ label ch30_minute(time_since_check):
 # NOTE: it only runs when the hour changes, so don't expect this to run
 #   on start right away
 label ch30_hour:
-    $ mas_runDelayedActions(MAS_FC_IDLE_HOUR)
-
-    #Runtime checks to see if we should have a consumable
-    $ MASConsumable._checkConsumables()
-
-    # clear ahoges if past noon
-    $ now_t = datetime.datetime.now().time()
-    if mas_isNtoSS(now_t) or mas_isSStoMN(now_t):
-        $ monika_chr._set_ahoge(None)
-
-    # xp calc
-    $ store.mas_xp.grant()
+    python:
+        mas_runDelayedActions(MAS_FC_IDLE_HOUR)
+
+        #Runtime checks to see if we should have a consumable
+        MASConsumable._checkConsumables()
+
+        # clear ahoges if past noon
+        now_t = datetime.datetime.now().time()
+        if mas_isNtoSS(now_t) or mas_isSStoMN(now_t):
+            monika_chr._set_ahoge(None)
+
+        # xp calc
+        store.mas_xp.grant()
+
+        #Set our TOD var
+        mas_setTODVars()
+
+        # Inc the chance for hold request
+        with MAS_EVL("monika_holdrequest") as holdme_ev:
+            # See if we flagged the ev
+            if holdme_ev.allflags(EV_FLAG_HFRS):
+                # Get time in hours the player has spent with Monika
+                hours_spent_this_sesh = int(round(float(mas_getSessionLength().total_seconds()) / 3600))
```

Comment:
is rounding important here? since we end up doing integer-based random checking later (1/x), seems like doing just integer truncation (which is rounding down) would be sufficient. No point in using `round` if the actual effect isn't needed.

If we really wanted the exact session time to influence the chance here, should just use total seconds out of 14400 (4*3600):
```python
chance = mas_getSessionLength().total_seconds() / (4*3600.0)
if chance < 0.2:
    chance = 0.2
    
if random.random() < chance:
     # unflag ev
```
^`random()` always returns between 0.0 and 1.0, so 4 hour+ session is handled since chance is greater than 1 in a 4+ hour session